jfrog-cli/artifactory/commands/buildinfo/addartifact.go

```diff
@@ -0,0 +1,70 @@
+package buildinfo
+
+import (
+	"errors"
+	"github.com/jfrogdev/jfrog-cli-go/jfrog-cli/artifactory/utils"
+	"github.com/jfrogdev/jfrog-cli-go/jfrog-cli/artifactory/spec"
+	"github.com/jfrogdev/jfrog-cli-go/jfrog-cli/utils/config"
+	"github.com/jfrogdev/jfrog-cli-go/jfrog-client/artifactory"
+	"github.com/jfrogdev/jfrog-cli-go/jfrog-client/utils/errorutils"
+	"github.com/jfrogdev/jfrog-cli-go/jfrog-client/utils/log"
+	"github.com/jfrogdev/jfrog-cli-go/jfrog-client/artifactory/buildinfo"
+	clientutils "github.com/jfrogdev/jfrog-cli-go/jfrog-client/artifactory/services/utils"
+)
+
+func AddArtifact(addArtifactSpec *spec.SpecFiles, flags *BuildAddArtifactConfiguration) (err error) {
+	log.Info("Adding artifacts to build info " + flags.BuildName + " #" + flags.BuildNumber + "...")
+
+	servicesManager, err := utils.CreateServiceManager(flags.ArtDetails, false); if err != nil {
+		return err
+	}
+
+	buildArtifacts, err := getBuildArtifacts(servicesManager, addArtifactSpec); if err != nil {
+		return err
+	}
+
+	err = utils.SaveBuildGeneralDetails(flags.BuildName, flags.BuildNumber); if err != nil {
+		return
+	}
+
+	populateFunc := func(partial *buildinfo.Partial) {
+		partial.Artifacts = buildArtifacts
+	}
+	err = utils.SavePartialBuildInfo(flags.BuildName, flags.BuildNumber, populateFunc); if err != nil {
+		return
+	}
+
+	log.Info("Successfully added artifact to build info")
+	return
+}
+
+func getBuildArtifacts(servicesManager *artifactory.ArtifactoryServicesManager, addArtifactSpec *spec.SpecFiles) ([]buildinfo.InternalArtifact, error) {
+	var buildArtifacts []buildinfo.InternalArtifact
+	for i := 0; i < len(addArtifactSpec.Files); i++ {
+		params, err := addArtifactSpec.Get(i).ToArtifatorySearchParams(); if err != nil {
+			return nil, err
+		}
+		resultItems, err := servicesManager.Search(&clientutils.SearchParamsImpl{ArtifactoryCommonParams: params}); if err != nil {
+			return nil, err
+		}
+		for _, resultItem := range resultItems {
+			artifactPath := resultItem.Repo + "/" + resultItem.Path + "/" + resultItem.Name
```

Comment:
I think there's a bug here, because *resultItem.Path* can have **.** as a value.
The *GetItemRelativePath* method of the *ResultItem* struct already takes care of it and should be probably used instead.
BTW,

Reply:
Overlooked this convenience function. It seems to return the expected path, so will use this instead.